app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,85 @@
+class Api::V1::CommentsController < ApplicationController
+  before_action :set_comment, only: %i[ show update destroy ]
+  include Authentication
+
+  # GET /comments
+  def index
+    comments = Comment.where(post_id: params[:post_id])
+
+    render json: comments
+  end
+
+  # GET /comments/1
+  def show
+    comment = Comment.where(post_id: params[:post_id], id: params[:id])
+
+    render json: comment 
+  end
+
+  # POST /comments
+  def create
+    if authenticate
+      post = Post.find(params[:post_id])
+      user = User.find_by(username: request.headers[:username])
```

Comment:
user is required at multiple places, so better add it in base controller and set the user in a before action.

Reply:
Created `get_user` to check if user exists and assign it to instance variable if it exists otherwise return 404 error